listscraper/utility_functions.py

```diff
@@ -42,4 +45,61 @@ def val2stars(val, not_found):
         stars = conv_dict[val]
         return stars
     except:
-        return not_found
\ No newline at end of file
+        return not_found
+
+def repeated_request(url):
+    """
+    Makes a request to a URL with a backoff for 429 errors.
+
+    Args:
+        url (str): The URL to make a request to.
+
+    Returns:
+        requests.Response: The successful response object.
+
+    Raises:
+        requests.exceptions.RequestException: If the request fails after all retries.
+    """
+    retry_seconds = [5,45,120]
+    retry_index = 0
+
+    headers = {
+        'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/115.0'
+    }
+
+    while retry_index < len(retry_seconds):
+        try:
+            response = requests.get(url, headers=headers)
```

Comment:
```suggestion
            response = requests.get(url, headers=headers)
            
```

newline b/w code & comment